File: app/src/main/java/com/amazon/tv/leanbacklauncher/notifications/NotificationUtils.java
package com.amazon.tv.leanbacklauncher.notifications;

import android.text.TextUtils;

import com.amazon.tv.tvrecommendations.TvRecommendation;

class NotificationUtils {
    public static boolean equals(TvRecommendation left, TvRecommendation right) {
        if (left == null || right == null) {
            return left == right;
        } else return TextUtils.equals(left.getKey(), right.getKey());
    }
}

File: app/src/main/java/com/amazon/tv/leanbacklauncher/notifications/NotificationUtils.kt
package com.amazon.tv.leanbacklauncher.notifications

import android.text.TextUtils
import com.amazon.tv.tvrecommendations.TvRecommendation

internal object NotificationUtils {
    @JvmStatic
    fun equals(left: TvRecommendation?, right: TvRecommendation?): Boolean {
        return if (left == null || right == null) {
            left == right
        } else left.key.equals(right.key)
    }
}